logfire/_internal/formatter.py

```diff
@@ -450,3 +451,29 @@ def warn_inspect_arguments(msg: str, stacklevel: int):
     ) + msg
     warnings.warn(msg, InspectArgumentsFailedWarning, stacklevel=stacklevel)
     logfire.log('warn', msg)
+
+
+class KnownFormattingError(Exception):
+    """An error raised when there's something wrong with a format string or the field values.
+
+    In other words this should correspond to errors that would be raised when using `str.format`,
+    and generally indicate a user error, most likely that they weren't trying to pass a template string at all.
+    """
+
+
+class FormattingFailedWarning(Warning):
+    pass
+
+
+def warn_formatting(msg: str):
+    _frame, stacklevel = get_user_frame_and_stacklevel()
+    warnings.warn(
+        f'\n'
+        f'    Ensure you are either:\n'
+        '      (1) passing an f-string directly, with inspect_arguments enabled and working, or\n'
+        '      (2) passing a literal `str.format`-style template, not a preformatted string.\n'
```

Comment:
You are also allowed to pass non-string-literals right? (By that I mean, a string-valued variable.) The way this message is phrased it sounds like that isn't allowed

Reply:
You're allowed, if you know what you're doing. On the one hand we used to have a `LiteralString` type hint for the argument, and it only got removed to allow passing f-strings. On the other hand the linked docs do suggest a reasonable way to pass a variable string. We could say 'low cardinality' but that's a bit heavy.

Reply:
That makes sense, given that I'm good with just leaving the message as is. 👍